src/main/java/mops/presentation/FileController.java

```diff
@@ -0,0 +1,92 @@
+package mops.presentation;
+
+import lombok.AllArgsConstructor;
+import mops.businesslogic.Account;
+import mops.businesslogic.FileService;
+import mops.businesslogic.utils.AccountUtil;
+import mops.persistence.file.FileInfo;
+import org.keycloak.adapters.springsecurity.token.KeycloakAuthenticationToken;
+import org.springframework.core.io.FileSystemResource;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.io.IOException;
+
+@Controller
+@RequestMapping("material1/file")
+@AllArgsConstructor
+public class FileController {
+
+    /**
+     * Handles actions on single files: Delete, Preview, Download.
+     */
+    private final FileService fileService;
+
+    /**
+     * @param token  a keycloak authentication token
+     * @param fileId the id of the requested file
+     * @return the route to template 'file'
+     */
+    @SuppressWarnings("PMD")
```

Comment:
why suppress PMD?

Reply:
Because.... PMD is never going to be happy with this method when using getter methods..